Onboarding page

jpan8866 opened the pull request:
## Description ✏️

- Refactor to use vscode globalState rather than file-based approach to determine first launch. 
- onboarding page in overlay (Hello + loading + open folder/file)

## Checklist ✅

- [ ] I have added screenshots (if UI changes are present).
- [ ] I have done a self-review of my code.
- [ ] I have manually tested my code (if applicable).

MaySoMusician commented on gui/src/pages/modelconfig.tsx:
```diff
@@ -25,7 +25,7 @@ import {
 const GridDiv = styled.div`
   display: grid;
-  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
+  grid-template-columns: 1fr;
```
Could you fix this prop too in `gui/src/pages/models.tsx`
It's also broken in the same way

MaySoMusician replied:
Ill deal with this myself

MaySoMusician commented on gui/src/pages/models.tsx:
```diff
@@ -49,7 +49,7 @@ function Models() {
           width="1.2em"
           height="1.2em"
-          onClick={() => navigate("/onboarding")}
+          onClick={() => navigate("/")}
```
Yeah thanks, but I should have commented as for this in more proper way

If a user arrive this page in onboarding process, this button should be navigate them to "/onboarding" page.
Otherwise, navigate them to "/" page.
I think we should pass some prop to change this value from outside

MaySoMusician replied:
https://github.com/trypear/pearai-submodule/pull/83#discussion_r1667669391

Himanshu-Singh-Chauhan replied:
> Yeah thanks, but I should have commented as for this in more proper way
> 
> If a user arrive this page in onboarding process, this button should be navigate them to "/onboarding" page. Otherwise, navigate them to "/" page. I think we should pass some prop to change this value from outside

yeah, thats why i left it as it is, where it navigates to changes based on what page it is right now.

jpan8866 committed:
Onboarding hello overlay + first launch globalstate

jpan8866 committed:
comment

jpan8866 committed:
working vers

jpan8866 committed:
undo parameters change

jpan8866 committed:
add welcome page

Himanshu-Singh-Chauhan committed:
new welcomepage/onboarding ui stuff (#99)

* color fixes and restart gif when card swithched

* gif switching fixes

* import extensions and add to paths steps

* update responsiveness

* final page

* final step signed in - open overlay on first launch - copy settings working - open folder working - logged in marks onboarding complete - social links

jpan8866 committed:
update toggle onboarding overlay logic flow

jpan8866 committed:
Merge branch 'panj/onboarding-hello' of https://github.com/trypear/pearai-submodule into panj/onboarding-hello

nang-dev committed:
nang backend plumbing for onboarding (#100)

* fix aider in quick pick (#89)

* fix aider in quick pick

* remove console

* Added hot reload and cleaned state for aider (#90)

* Added hot reload and fixed state

* Added return type

---------

Co-authored-by: nang-dev <contactnang88@gmail.com>

* Aider context providers fix in overlay (#92)

* added pearai_aider needlessly

* prrrroogress

* Added working

* Added working

* Back to pearai_model for aider

* Added cleanup

* Fixed bug

---------

Co-authored-by: nang-dev <contactnang88@gmail.com>

* uiux improvements (#88)

* show loading on quickinput

* remove add more context providers from creator mode

* perplexity style streaming (working in sidebar)

* working perplexity like streaming in perplexity and aider

* rerenders and jump fixes after adding fade in perplexity style streaming

* fade in for headings, ul, ol, li (lists)

* remove add more context providers from perplexity

---------

Co-authored-by: Nang <nathanang2000@gmail.com>

* Added exclusions  (#93)

* Added exclusions

* Added exclusions

* Added exclusions

---------

Co-authored-by: nang-dev <contactnang88@gmail.com>

* Added fix (#94)

Co-authored-by: nang-dev <contactnang88@gmail.com>

* Aider fix (#96)

* Added progress

* Added restart button to aider

---------

Co-authored-by: nang-dev <contactnang88@gmail.com>

* Aider fix 2 (#97)

* Added progress

* Added restart button to aider

* Added restart button to aider

---------

Co-authored-by: nang-dev <contactnang88@gmail.com>

* Added provider name for metrics (#98)

Co-authored-by: nang-dev <contactnang88@gmail.com>

* Added progress

* Added refactor

* Added progress

* Added progress

* Revert "Merge branch 'main' of https://github.com/trypear/pearai-submodule into nang/onboarding-hello"

This reverts commit 4dcd7c3051076cf47342a229148bfecada5b3469, reversing
changes made to 41319e51c2b8d5a47254e8062420267638ccc2c1.

* Actually working now

* Progress, sign in is messed up
'

* Added parity

* Added add to path

* Added path functionality

* Added better wording

* Added login cleanup

* Added login cleanup

* Fixed login trigger

* Cleaned up

* Cleaned up

---------

Co-authored-by: John Pan <john.pan@mail.mcgill.ca>
Co-authored-by: nang-dev <contactnang88@gmail.com>
Co-authored-by: Himanshu <him.speedo@gmail.com>

nang-dev2 committed:
Merge branch 'main' of https://github.com/trypear/pearai-submodule into panj/onboarding-hello

nang-dev2 committed:
Fixed typo

Himanshu-Singh-Chauhan committed:
lock overlay on onboarding

Himanshu-Singh-Chauhan committed:
remove file based firstLanchFlag

Himanshu-Singh-Chauhan committed:
liniting

Himanshu-Singh-Chauhan committed:
conflict resolving with main (merge main)

Himanshu-Singh-Chauhan committed:
no backing to import ext page

Himanshu-Singh-Chauhan committed:
fixes

Himanshu-Singh-Chauhan committed:
fixes

Himanshu-Singh-Chauhan committed:
welcome page only on firstlaunch

Himanshu-Singh-Chauhan committed:
cleanup
